build.rs

```diff
@@ -119,41 +121,120 @@ fn main() {
 
     // C header search paths
     let mut header_search_paths: Vec<PathBuf> = Vec::new();
-    let target_os = env_var("CARGO_CFG_TARGET_OS");
 
     if cfg!(feature = "use_system_capstone") {
-        #[cfg(feature = "use_system_capstone")]
+        #[cfg(windows)] panic!("Feature 'use_system_capstone' is not supported on Windows");
+        #[cfg(all(not(windows), feature = "use_system_capstone"))]
         {
             link_type = find_system_capstone(&mut header_search_paths);
         }
     } else {
         if cfg!(feature = "build_capstone_cmake") {
-            #[cfg(feature = "build_capstone_cmake")] cmake();
+            #[cfg(windows)] panic!("Feature 'build_capstone_cmake' is not supported on Windows");
+            #[cfg(all(not(windows), feature = "build_capstone_cmake"))] cmake();
         } else {
-            // TODO: need to add this argument for windows 64-bit, msvc, dunno, read
-            // COMPILE_MSVC.txt file cygwin-mingw64
-
-            // In BSDs, `make` does not refer to GNU make
-            let make_cmd = if target_os.contains("bsd") || target_os == "dragonfly" {
-                "gmake"
-            } else {
-                "make"
-            };
-
-            let out_dir = env_var("OUT_DIR");
-            Command::new(make_cmd)
-                .current_dir(CAPSTONE_DIR)
-                .status()
-                .expect("Failed to build Capstone library");
-            let capstone = "libcapstone.a";
-            Command::new("cp")
-                .current_dir(CAPSTONE_DIR)
-                .arg(&capstone)
-                .arg(&out_dir)
-                .status()
-                .expect("Failed to copy capstone library to OUT_DIR");
-
-            println!("cargo:rustc-link-search=native={}", out_dir);
+            #[cfg(windows)]
+            {
```

Comment:
Could we refactor this block into a separate function? Something like `build_capstone_cc()`?

(also see comment below for other refactor)